propagation.go

```diff
@@ -115,9 +115,9 @@ func (p *textMapPropagator) Inject(
 	// Do not encode the string with trace context to avoid accidental double-encoding
 	// if people are using opentracing < 0.10.0. Our colon-separated representation
 	// of the trace context is already safe for HTTP headers.
-	textMapWriter.Set(TracerStateHeaderName, sc.String())
+	textMapWriter.Set(p.tracer.headerKeys.TracerStateHeaderName, sc.String())
```

Comment:
not the biggest fan of `p.tracer.headerKeys.etc`; looking at the code, it looks like tracer is passed to all these propagators but it's never used (except one place where it's used just for its metrics). It'll look cleaner if we pass in headerkeys (and metrics where warranted) only instead of the whole tracer.